cmux.go

```diff
@@ -177,7 +178,15 @@ func (m *cMux) handleErr(err error) bool {
 
 type muxListener struct {
 	net.Listener
-	connc chan net.Conn
+	connc  chan net.Conn
+	closed *sync.Once
```

Comment:
Unfortunately it can't because the muxListener gets passed around by value, and copying a sync.Once object around is a no-no.